uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -0,0 +1,220 @@
+"""Tests related to uaclient.entitlement.base module."""
+
+import copy
+from io import StringIO
+import mock
+from nose2.tools.params import params
+
+from uaclient import config
+from uaclient.entitlements.livepatch import (
+    LivepatchEntitlement, process_directives)
+from uaclient import status
+from uaclient.testing.helpers import TestCase
+
+
+LIVEPATCH_MACHINE_TOKEN = {
+    'machineSecret': 'blah',
+    'machineTokenInfo': {
+        'contractInfo': {
+            'resourceEntitlements': [
+                {'type': 'livepatch'}]}}}
+
+
+LIVEPATCH_RESOURCE_ENTITLED = {
+    'resourceToken': 'TOKEN',
+    'entitlement': {
+        'obligations': {
+            'enableByDefault': False
+        },
+        'type': 'livepatch',
+        'entitled': True,
+        'directives': {
+            'caCerts': '',
+            'remoteServer': 'https://alt.livepatch.com'
+        },
+        'affordances': {
+            'kernelFlavors': ['generic', 'lowlatency'],
+            'tier': 'stable'
+        }
+    }
+}
+
+M_PATH = 'uaclient.entitlements.livepatch.'  # mock path
+
+
+class TestLivepatchContractStatus(TestCase):
```

Comment:
We could reduce some of the repetition in several of these test classes by adding a super-class with a `setUp` that does:

```py
tmp_dir = self.tmp_dir()
self.cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
self.cfg.write_cache('machine-token', LIVEPATCH_MACHINE_TOKEN)
self.cfg.write_cache('machine-access-livepatch', LIVEPATCH_RESOURCE_ENTITLED)
```

Reply:
Good suggestion added BaseEnabledLivepatchConfig class for that setup